Modules/_sha3/sha3module.c

```diff
@@ -746,8 +735,26 @@ PyInit__sha3(void)
         goto error;
     }
 
-    return m;
+    return 0;
   error:
-    Py_DECREF(m);
-    return NULL;
+    return -1;
 }
+
+static PyModuleDef_Slot _sha3_slots[] = {
+    {Py_mod_exec, sha3_exec},
+    {0, NULL}
+};
+
+/* Initialize this module. */
+static struct PyModuleDef _sha3module = {
+        PyModuleDef_HEAD_INIT,
+        .m_name = "_sha3",
+        .m_size = sizeof(SHA3State),
```

Comment:
If you add a state with heap allocated states, it would be nice to add traverse, clear and free functions. See for example the _abc module.

Reply:
@vstinner thanks I added this to my list of things to look for.